### PR TITLE
Fix for "one word aliases of categories treated as articles" with adv…

### PR DESCRIPTION
**Pull request by alex7r**
#### Summary of Changes

Include advanced SEF enabled check
#### Testing Instructions

Enable advanced SEF
Create Category
Create Sub-category (one word alias)
Create article in sub-category

Link menu Item to Category
#### Bug result

Category can be accessed
Article in sub-category can be accessed
Sub-category can't be accessed because it treated as article(while having one word alias) and article not found
#### Fix result

Category can be accessed
Sub-category can be accessed
Articles of all levels can be accessed

### CODE REVIEW

**Comment by wojsmol** [comment]
@alex7r Please see https://github.com/alex7r/joomla-cms/pull/1

**Comment by alex7r** [comment]
Passed checks with [Build #23692](https://travis-ci.org/joomla/joomla-cms/builds/128338101 "Build #23692")<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10261">issues.joomla.org/joomla-cms/10261</a>.</sub>

**Comment by wilsonge** [comment]
https://github.com/joomla/joomla-cms/blob/3.6.x/components/com_content/router.php

We have totally rewritten this in the 3.7 branch there. The advanced router is not a documented feature so I'm going to close this PR as its not a usable feature and it's being removed soon anyhow

**Comment by alex7r** [comment]
It may be not documented, but planing on removing it - is like planing on giving Joomla big disadvantage.
It's a good idea if you want to maintain value of SEF extensions wich making Joomla sites scare people and blocking Joomla community from it's growth.
Really, advanced SEF is mast have feature and it's working.(working on a live site, 3000+ visits per day about an year) There little problem with triggering 404 page, but it can be managed by checking  current URL and distanation URL to match(can provide my code).
And if you think that standard routing is what people need - no, SEO asks to exclude 'id-' from URL.
So to summarize: cut out advanced SEF - give people one more reason to say "oh, joomla? I'm out".
Joomla must be for everyone, not for "everyone who don't bother about his site".

p.s. sorry for "fire in the hole" on this one, but each talk with people who prefer anything against Joomla is like knife in heart for me. And you planning to give this people one more real reason to refuse to show love to J!
You can email me for more details on "why SEF extensions are bad" and "what people need to have in core" and any other feedback.

Thanks. And don't cut it off...

**Comment by wilsonge** [comment]
We have https://github.com/joomla/joomla-cms/pull/10170 as an alternative for removing ids from URLs

**Comment by alex7r** [comment]
oh, well, if it gonna  make it into, then it's okay
mean it's okay if in the core there will be way to do it
thanks for clarifying this for me

**Comment by wilsonge** [comment]
It's ok :) Advanced routing was basically an experiment for 1.6 that never got finished by Hannes. This is the actual production ready code :)
